### Business Data Analytics and Prediction
### Dr. Gilli Shama 
# Lesson 1.1 Introduction 

In this lesson we will read the HR data and view it, see its summary to understand the columns and to learn what characterises those who left the company, and check that there are no missing values.


## Read the data 

The business analytics concepts and code will be demonstrated on HR data of employees attrition, evaluation and satisfaction: https://www.kaggle.com/ludobenistant/hr-analytics

Download the Excel file from the link above, to your local folder, and then read it to a data frame, as below.


In [1]:
import pandas as pd #pandas is the basic data analytics library, which enables to handle data frames

In [2]:
df = pd.read_excel("C:/Users/gillis/Documents/_Personal/IDC/Business_Data_Analytics/human-resources-analytics/HR_comma_sep.xlsx")  #This is the data file we reviewed previously in Tableau

Files can be read as csv. However, some csv files require the encoding type. In this case you will need to check your file encoding first. This check can be done with note++.

If your original file is csv and you save it as Excel, then some data might be lost without any warning, incase of very long files, or cells with very long text.

## Display data 

To view the entire data frame, as View in R, just click the data name.
Other commands are very similar to R.

In [3]:
len(df)  #Shows the length, that is number of records

14999

In [4]:
df.head()  # Shows the first records, same as header command in python

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
1                0.80             0.86               5                   262   
2                0.11             0.88               7                   272   
3                0.72             0.87               5                   223   
4                0.37             0.52               2                   159   

   time_spend_company  Work_accident  left  promotion_last_5years  sales  \
0                   3              0     1                      0  sales   
1                   6              0     1                      0  sales   
2                   4              0     1                      0  sales   
3                   5              0     1                      0  sales   
4                   3              0     1                      0  sales   

   salary  
0     low  
1  medium  
2  medium  
3     low  
4     low

In [5]:
df.info()  # Same as str command in R

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 10 columns):
satisfaction_level       14999 non-null float64
last_evaluation          14999 non-null float64
number_project           14999 non-null int64
average_montly_hours     14999 non-null int64
time_spend_company       14999 non-null int64
Work_accident            14999 non-null int64
left                     14999 non-null int64
promotion_last_5years    14999 non-null int64
sales                    14999 non-null object
salary                   14999 non-null object
dtypes: float64(2), int64(6), object(2)
memory usage: 1.1+ MB


## Summary statistics of data 

To get basic statistics of the data use the describe function.

In [6]:
df.describe()  # Same as Summary command in R

satisfaction_level  last_evaluation  number_project  \
count        14999.000000     14999.000000    14999.000000   
mean             0.612834         0.716102        3.803054   
std              0.248631         0.171169        1.232592   
min              0.090000         0.360000        2.000000   
25%              0.440000         0.560000        3.000000   
50%              0.640000         0.720000        4.000000   
75%              0.820000         0.870000        5.000000   
max              1.000000         1.000000        7.000000   

       average_montly_hours  time_spend_company  Work_accident          left  \
count          14999.000000        14999.000000   14999.000000  14999.000000   
mean             201.050337            3.498233       0.144610      0.238083   
std               49.943099            1.460136       0.351719      0.425924   
min               96.000000            2.000000       0.000000      0.000000   
25%              156.000000            3.000000       0.000000      0.000000   
50%              200.000000            3.000000       0.000000      0.000000   
75%              245.000000            4.000000       0.000000      0.000000   
max              310.000000           10.000000       1.000000      1.000000   

       promotion_last_5years  
count           14999.000000  
mean                0.021268  
std                 0.144281  
min                 0.000000  
25%                 0.000000  
50%                 0.000000  
75%                 0.000000  
max                 1.000000

We can see that “left” column is an integer with values of either 0 or 1, where 24% of the employees are 1, that is left the company. Promotion and Work Accident variables are also 0 or 1. Only 2% recived a promotion in last 5 years. Work accident happened to 14% of the emloyees.